Multiple CacheStatisticsMXBeanImpl as Leftover MBeans failures

- Fails on `Hazelcast-4.master-CorrettoJDK8`
- Fails on [Build #88 (Sep 2, 2020 6:15:00 AM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.master-CorrettoJDK8/88/testReport/junit/com.hazelcast.cache/HazelcastServerCachingProviderTest/com_hazelcast_cache_HazelcastServerCachingProviderTest/)
- Error
```
Leftover MBeans are still registered with the platform MBeanServer: [com.hazelcast.cache.impl.CacheStatisticsMXBeanImpl[javax.cache:type=CacheStatistics,CacheManager=,Cache=cache1]]
```
- Stacktrace
```
java.lang.AssertionError: Leftover MBeans are still registered with the platform MBeanServer: [com.hazelcast.cache.impl.CacheStatisticsMXBeanImpl[javax.cache:type=CacheStatistics,CacheManager=,Cache=cache1]]
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.cache.jsr.JsrTestUtil.assertNoMBeanLeftovers(JsrTestUtil.java:209)
	at com.hazelcast.cache.jsr.JsrTestUtil.setup(JsrTestUtil.java:57)
	at com.hazelcast.cache.HazelcastServerCachingProviderTest.init(HazelcastServerCachingProviderTest.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:301)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
....
```
- Similar failure for `HazelcastClientCachingProviderTest `
```
Leftover MBeans are still registered with the platform MBeanServer: [com.hazelcast.cache.impl.CacheStatisticsMXBeanImpl[javax.cache:type=CacheStatistics,CacheManager=hazelcast,Cache=cache-B], com.hazelcast.cache.impl.CacheStatisticsMXBeanImpl[javax.cache:type=CacheStatistics,CacheManager=hazelcast,Cache=cache-A], com.hazelcast.cache.impl.CacheStatisticsMXBeanImpl[javax.cache:type=CacheStatistics,CacheManager=,Cache=cache1]]
```

- Causing [CacheCreateUseDestroyTest](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.master-Solaris-OracleJDK8/217/com.hazelcast$hazelcast/testReport/com.hazelcast.cache.impl/CacheCreateUseDestroyTest/testCache_whenDestroyedByCacheManager_OBJECT_/) to fail on `Hazelcast-4.master-Solaris-OracleJDK8`
- Causing [MergePolicyValidatorCachingProviderIntegrationTest](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.master-Solaris-OracleJDK8/217/com.hazelcast$hazelcast/testReport/com.hazelcast.internal.config/MergePolicyValidatorCachingProviderIntegrationTest/com_hazelcast_internal_config_MergePolicyValidatorCachingProviderIntegrationTest/) to fail on `Hazelcast-4.master-Solaris-OracleJDK8`
- It is also observed many other builds. i.e. `Hazelcast-4.master-sonar`
- It is also causing failures on `EE` side. [Sample on EE](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-EE-4.master-sonar/587/testReport/com.hazelcast.cache/EnterpriseCacheCreationTest/createSingleCache/)
- kindly check

--- a/hazelcast/src/test/java/com/hazelcast/cache/merge/CacheEventListenerSplitBrainTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cache/merge/CacheEventListenerSplitBrainTest.java
@@ -97,7 +97,6 @@ public class CacheEventListenerSplitBrainTest extends SplitBrainTestSupport {
         if (inMemoryFormat == NATIVE) {
             cacheConfig.getEvictionConfig().setMaxSizePolicy(USED_NATIVE_MEMORY_SIZE);
         }
-        cacheConfig.setStatisticsEnabled(true);
         cacheConfig.getMergePolicyConfig().setPolicy(mergePolicyClass.getName());
 
         if (updatedListener != null) {
@@ -170,12 +169,7 @@ public class CacheEventListenerSplitBrainTest extends SplitBrainTestSupport {
     }
 
     private void assert_update_event_generated_on_merge_of_not_equal_entries() {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals(1, cacheBUpdatedListener.updated.get());
-            }
-        });
+        assertTrueEventually(() -> assertEquals(1, cacheBUpdatedListener.updated.get()));
     }
 
     static class TestCacheEntryUpdatedListener<K, V> implements CacheEntryUpdatedListener<K, V>, Serializable {

--- a/hazelcast/src/test/java/com/hazelcast/cache/stats/CacheStatsTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cache/stats/CacheStatsTest.java
@@ -751,9 +751,14 @@ public class CacheStatsTest extends CacheTestSupport {
         // this put must trigger eviction
         cache1.put(key, "foo");
 
-        // number of evictions on primary and backup must be 1
-        assertEquals(1, cache1.getLocalCacheStatistics().getCacheEvictions()
-                + cache2.getLocalCacheStatistics().getCacheEvictions());
+        try {
+            // number of evictions on primary and backup must be 1
+            assertEquals(1, cache1.getLocalCacheStatistics().getCacheEvictions()
+                    + cache2.getLocalCacheStatistics().getCacheEvictions());
+        } finally {
+            cache1.destroy();
+            cache2.destroy();
+        }
     }
 
     @Test(expected = UnsupportedOperationException.class)

